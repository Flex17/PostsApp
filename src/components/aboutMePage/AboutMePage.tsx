import React from "react";

const AboutMePage: React.FC = () => {

    return (
        <div data-testid="about-me-elem">
        	<div className="fs-4 mb-4 mt-3">
				Здравствуйте, меня зовут Владислав!
			</div>
			<div className="fs-5 mb-2">
				Мне 18 лет, учусь на 1 курсе Тюменского государственного университета, специальность: Информационные системы и технологии.<br/> Проживаю, соответственно, тоже в городе Тюмень.
			</div>
			<div className="fs-5 mb-2">
				Фронтендом начал заниматься примерно 3 года тому назад, путь был такой: верстка &#8594; js &#8594; react &#8594; ts <br/>
				Уже успел поработать 1 год в компании Smart Bit, меня взяли на проект wakadoo.ru, можно сказать, что я один писал весь фронт (авторизация, работа с api, разрешения, роли и тд), разве что на первых порах мне помогал наставник. <br/>
				Сейчас этот проект практически закончен, дорабатываются некоторые мелочи, но я к нему уже не причастен.
			</div>
			<div className="fs-5 mb-4">
				Отмечу, что было интересно делать ваше тестовое задание, был приятно удивлен тз и тем, насколько оно подробно написано.<br/>
				В завершении скажу, что хотел бы работать в вашей компании, с нетерпением жду ответу и желательно комментариев по тестовому заданию, было бы интересно их услышать!
			</div>
			<div className="fs-5 text-end">
				Спасибо за уделенное время!
			</div>
        </div>
    );
};

export default AboutMePage;
