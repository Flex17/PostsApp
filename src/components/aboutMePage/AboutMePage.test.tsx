import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import AboutMePage from "./AboutMePage";

describe('About me page component', () => {
    test('Component rendered', () => {
        render(<AboutMePage />);

        expect(screen.getByText('Здравствуйте, меня зовут Владислав!')).toBeInTheDocument();
        expect(screen.getByText('Мне 18 лет, учусь на 1 курсе Тюменского государственного университета, специальность: Информационные системы и технологии. Проживаю, соответственно, тоже в городе Тюмень.')).toBeInTheDocument();
        expect(screen.getByText('Фронтендом начал заниматься примерно 3 года тому назад, путь был такой: верстка → js → react → ts Уже успел поработать 1 год в компании Smart Bit, меня взяли на проект wakadoo.ru, можно сказать, что я один писал весь фронт (авторизация, работа с api, разрешения, роли и тд), разве что на первых порах мне помогал наставник. Сейчас этот проект практически закончен, дорабатываются некоторые мелочи, но я к нему уже не причастен.')).toBeInTheDocument();
        expect(screen.getByText('Спасибо за уделенное время!')).toBeInTheDocument();
    });

    test('About me page snapshot', () => {
        const page = render(<AboutMePage/>);

        expect(page).toMatchSnapshot();
    })
});

