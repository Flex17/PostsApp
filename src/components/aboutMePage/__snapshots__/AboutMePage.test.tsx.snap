// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About me page component About me page snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="about-me-elem"
      >
        <div
          class="fs-4 mb-4 mt-3"
        >
          Здравствуйте, меня зовут Владислав!
        </div>
        <div
          class="fs-5 mb-2"
        >
          Мне 18 лет, учусь на 1 курсе Тюменского государственного университета, специальность: Информационные системы и технологии.
          <br />
           Проживаю, соответственно, тоже в городе Тюмень.
        </div>
        <div
          class="fs-5 mb-2"
        >
          Фронтендом начал заниматься примерно 3 года тому назад, путь был такой: верстка → js → react → ts 
          <br />
          Уже успел поработать 1 год в компании Smart Bit, меня взяли на проект wakadoo.ru, можно сказать, что я один писал весь фронт (авторизация, работа с api, разрешения, роли и тд), разве что на первых порах мне помогал наставник. 
          <br />
          Сейчас этот проект практически закончен, дорабатываются некоторые мелочи, но я к нему уже не причастен.
        </div>
        <div
          class="fs-5 mb-4"
        >
          Отмечу, что было интересно делать ваше тестовое задание, был приятно удивлен тз и тем, насколько оно подробно написано.
          <br />
          В завершении скажу, что хотел бы работать в вашей компании, с нетерпением жду ответу и желательно комментариев по тестовому заданию, было бы интересно их услышать!
        </div>
        <div
          class="fs-5 text-end"
        >
          Спасибо за уделенное время!
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="about-me-elem"
    >
      <div
        class="fs-4 mb-4 mt-3"
      >
        Здравствуйте, меня зовут Владислав!
      </div>
      <div
        class="fs-5 mb-2"
      >
        Мне 18 лет, учусь на 1 курсе Тюменского государственного университета, специальность: Информационные системы и технологии.
        <br />
         Проживаю, соответственно, тоже в городе Тюмень.
      </div>
      <div
        class="fs-5 mb-2"
      >
        Фронтендом начал заниматься примерно 3 года тому назад, путь был такой: верстка → js → react → ts 
        <br />
        Уже успел поработать 1 год в компании Smart Bit, меня взяли на проект wakadoo.ru, можно сказать, что я один писал весь фронт (авторизация, работа с api, разрешения, роли и тд), разве что на первых порах мне помогал наставник. 
        <br />
        Сейчас этот проект практически закончен, дорабатываются некоторые мелочи, но я к нему уже не причастен.
      </div>
      <div
        class="fs-5 mb-4"
      >
        Отмечу, что было интересно делать ваше тестовое задание, был приятно удивлен тз и тем, насколько оно подробно написано.
        <br />
        В завершении скажу, что хотел бы работать в вашей компании, с нетерпением жду ответу и желательно комментариев по тестовому заданию, было бы интересно их услышать!
      </div>
      <div
        class="fs-5 text-end"
      >
        Спасибо за уделенное время!
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
